### PR TITLE
ENH cope with API change

### PR DESCRIPTION
**Pull request by tomMoral**
Parameter `INFINITY` has been move in the refactoring benchopt/benchopt#265.

This PR intends to avoid leaving it in `runner.py` while it is not needed anymore.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
     env:
       CONDA_ENV: 'test_env'
       VERSION_PYTHON: ${{ matrix.version_python }}
-      BENCHOPT_BRANCH: benchopt:main
+      BENCHOPT_BRANCH: tommoral:ENH_parallelization
```
Comment (mathurinm) [inline .github/workflows/main.yml:28]: is this needed for this PR or committed by accident ?

File: datasets/libsvm.py [hunk 2]
```diff
@@ -20,13 +20,11 @@ class Dataset(BaseDataset):
 
     def __init__(self, dataset="bodyfat"):
         self.dataset = dataset
-        self.X, self.y = None, None
```
Comment (mathurinm) [inline datasets/libsvm.py:23]: I think this file should not be touched

File: .github/workflows/main.yml (hunk 1 above)
Comment (tomMoral) [reply .github/workflows/main.yml:28]: This is a mechanism to run the test with a branch that is not `main`. As this PR depends on this change, this is intentionnal.

File: datasets/libsvm.py (hunk 2 above)
Comment (tomMoral) [reply datasets/libsvm.py:23]: Actually, the change in #255 is that all the dataset direclty implement a caching mecahnism so this part is not needed anymore.

File: .github/workflows/main.yml (hunk 1 above)
Comment (tomMoral) [reply .github/workflows/main.yml:28]: As benchopt/benchopt#255 has been merged, we can now revert this
```suggestion
      BENCHOPT_BRANCH: benchopt:main
```

**Pull request merged**